COSC481W-2024Fall/Ript-Fitness
# Socialfeed / Homepage

# Social Feed / Homepage

## Description:

This PR implements the initial version of the Social Feed. The feed is the default home screen that users will see after logging in and allows users to view text and image posts from their friends, keeping them motivated and connected.

## Key Changes:

- Implemented the home screen that displays user-generated posts (both text and image posts).
- Posts are displayed in chronological order, with the newest posts appearing first.
- Users can interact with the feed by liking posts.
- Added a pull-to-refresh functionality to reload the posts.
- Integrated basic Haptic feedback for interactions (likes).
- Handled the scenario where no posts are available by showing a placeholder message.
- Responsive design is ensured, so the layout adjusts appropriately across screen sizes.


## What’s Not Completed:

- The streak counter which was mentioned in the git issue belongs to a separate git issue
- Data fetching from the database is currently mocked. Integration with the backend to pull real data will be completed in a future sprint.


## Testing Instructions:

- Navigate to the Social Feed after logging in (once merged with login page). 
- Posts should display in chronological order.
- Like any post by tapping the heart icon. The heart icon will turn red to indicate the post is liked, and haptic feedback will occur. Pull down on the screen to refresh the posts. 
- Verify that the layout adjusts appropriately on different screen sizes and handles both text and image posts.

Issue #7
